exchange/src/exchange/traders/trader_manager.cpp

```diff
@@ -6,12 +6,11 @@ namespace manager {
 std::optional<messages::SIDE>
 ClientManager::validate_match(const messages::Match& match) const
 {
-    float trade_value = match.price * match.quantity;
-    float remaining_capital = get_capital(match.buyer_id) - trade_value;
+    double trade_value = match.price * match.quantity;
+    double remaining_capital = get_trader(match.buyer_id)->get_capital() - trade_value;
```

Comment:
you really should store pointers to the traders in your match this is way slower